BE/src/main/java/kr/codesquad/sidedish/service/OrderService.java

```diff
@@ -0,0 +1,36 @@
+package kr.codesquad.sidedish.service;
+
+import kr.codesquad.sidedish.domain.Item;
+import kr.codesquad.sidedish.domain.Order;
+import kr.codesquad.sidedish.repository.ItemRepository;
+import kr.codesquad.sidedish.repository.OrderRepository;
+import kr.codesquad.sidedish.web.dto.order.OrderRequest;
+import kr.codesquad.sidedish.web.dto.order.OrderResponse;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class OrderService {
+
+    private final OrderRepository orderRepository;
+    private final ItemRepository itemRepository;
+
+    public OrderService(OrderRepository orderRepository, ItemRepository itemRepository) {
+        this.orderRepository = orderRepository;
+        this.itemRepository = itemRepository;
+    }
+
+    public OrderResponse create(OrderRequest orderRequest) {
+        Order saved = orderRepository.save(orderRequest.toEntity());
+        Integer quantity = orderRequest.getQuantity();
+        Long itemId = orderRequest.getItemId();
+        Optional<Item> orderItem = itemRepository.findById(itemId);
+        Item item = orderItem.orElseThrow();
+        Integer stock = item.getStock();
+        int newStock = stock - quantity;
+
+        orderRepository.updateStock(newStock, itemId);
```

Comment:
item 객체의 stock 변경이 service 로직에서 이루어지기보다는 item 객체 내부에서 메서드를 생성하고 처리하는 편이 좀 더 객체지향스러운 설계가 될 것 같습니다. 아래와 같은 느낌으로요.

```suggestion
        item.changeStock(quantity); // 내부에서 stock - quantity 등 진행
        orderRepository.updateStock(item.getStock(), itemId);
```

Reply:
네 헨리! 해당 로직이 `Service`단에서 처리해도 무난할것 같다고 생각했었는데
말씀주신대로 `item` 객체 내부에 로직을 숨기니 보다 깔끔한 코드가 된것 같습니다. 감사합니다. 🙆‍♂️